client/checkout/classic/index.js

```diff
@@ -44,6 +44,11 @@ jQuery( function ( $ ) {
 	// event for this. This part of the page can also reload based on changes to checkout details, so we call unmount
 	// first to ensure the card element is re-mounted correctly.
 	$( document.body ).on( 'updated_checkout', () => {
+		// If the card element selector doesn't exist, then do nothing (for example, when a 100% discount coupon is applied).
+		if ( ! $( '#wcpay-card-element' ).length ) {
```

Comment:
I'd suggest to merge this into next condition: `if ( ! $( '#wcpay-card-element' ).length || $( '#wcpay-card-element' ).children().length)`, as they guarding the element mounting.

Reply:
That makes sense. I've updated the PR in 7cb98b161ad5cf715847c561066aa4c0c84f0fa2.